internal/backend/rest/rest.go

```diff
@@ -173,7 +174,29 @@ func (b *Backend) IsNotExist(err error) bool {
 // Load runs fn with a reader that yields the contents of the file at h at the
 // given offset.
 func (b *Backend) Load(ctx context.Context, h restic.Handle, length int, offset int64, fn func(rd io.Reader) error) error {
-	return backend.DefaultLoad(ctx, h, length, offset, b.openReader, fn)
+	r, err := b.openReader(ctx, h, length, offset)
+	if err != nil {
+		return err
+	}
+	err = fn(r)
+
+	// Note: readerat.ReadAt() (the fn) uses io.ReadFull() that doesn't
+	// wait for EOF after reading body. Due to HTTP/2 stream multiplexing
+	// and goroutine timings the EOF frame arrives from server (eg. rclone)
+	// with a delay after reading body. Immediate close might trigger
+	// HTTP/2 stream reset resulting in the *stream closed* error on server,
+	// so we wait for EOF before closing body.
+	var buf [1]byte
+	if n, e := r.Read(buf[:]); n != 0 || e != io.EOF {
+		fmt.Fprintf(os.Stderr, "%v: error waiting for EOF from server\n", h)
```

Comment:
Just logging it to `stderr` does not handle the error and will mess with other programs calling restic.

We need to figure out what to do with the error returned by `Read()`:
 * Either return it from `Load()` and pass it up the stack
 * Or ignore it

IMHO, we can drop the check for `n` and just call `Read()` and inspect the error: If it's `nil` or `io.EOF` then ignore it, otherwise return it and pass it to the caller. Thoughts?

Reply:
done